sources/packages/web/src/services/http/DesignationAgreementApi.ts

```diff
@@ -22,13 +26,39 @@ export class DesignationAgreementApi extends HttpBaseClient {
     designationId: number,
   ): Promise<GetDesignationAgreementDto> {
     return this.getCallTyped<GetDesignationAgreementDto>(
-      `institution/designation-agreement/${designationId}`,
+      this.addClientRoot(`designation-agreement/${designationId}`),
     );
   }
 
-  async getDesignationsAgreements(): Promise<GetDesignationAgreementsDto[]> {
+  async getDesignationsAgreements(
+    institutionId?: number,
+  ): Promise<GetDesignationAgreementsDto[]> {
+    const url = institutionId
+      ? `designation-agreement/institution/${institutionId}`
+      : "designation-agreement";
     return this.getCallTyped<GetDesignationAgreementsDto[]>(
-      `institution/designation-agreement`,
+      this.addClientRoot(url),
+    );
+  }
+
+  async getDesignationByStatus(
+    designationStatus: DesignationAgreementStatus,
+    searchCriteria?: string,
+  ): Promise<PendingDesignationDto[]> {
+    let url = `designation-agreement/status/${designationStatus}`;
+    if (searchCriteria) {
+      url = `${url}?${PaginationParams.SearchCriteria}=${searchCriteria}`;
+    }
+    return this.getCallTyped<PendingDesignationDto[]>(this.addClientRoot(url));
```

Comment:
is await not needed here ?

Reply:
I believe it should be fine as await is used at call of this service method and also inside getCallTyped.